Assets/Scripts/PogoControls.cs

```diff
@@ -249,4 +278,39 @@ private void OnDrawGizmos()
         // Draw the pogo stick center
         Gizmos.DrawSphere(pogoStick.transform.position + pogoStick.transform.rotation * flipAxisOffset, 4);
     }
+
+
+    // Magic trick functions
+    //TODO: maybe consider moving this to a separate script
+    public void WaterSpell()
+    {
+        //spawn ice effect
+        GameObject effect = Instantiate(iceEffect, transform.position, Quaternion.identity);
+        //destroy the ice effect after 1 second
+        Destroy(effect, 1);
+
+        //if player is on a water layer, turn the water into ice
+        LayerMask layerMask = LayerMask.GetMask("Water");
+        RaycastHit hit;
+        Vector3 pogoCastStart = leanChild.transform.position + leanChild.transform.rotation * pogoRayCastOffset;
+        Vector3 pogoCastEnd = pogoCastStart + leanChild.transform.rotation * leanChild.transform.up * (-pogoRayCastLength);
+        if (Physics.Raycast(pogoCastStart, -1 * leanChild.transform.up, out hit, pogoRayCastLength, layerMask))
+        {   
+            //turn water into ice
+            hit.collider.gameObject.layer = 0;
+            Debug.Log("layer now: " + hit.collider.gameObject.layer);
+            //change the material of the water to ice
+            hit.collider.gameObject.GetComponent<Renderer>().material.SetColor("_Color", Color.white);
+        }
+    }
+
+    public void FireSpell()
```

Comment:
Eventually, we'll only want to trigger the jump if the spell detonates an explosive.